### PR TITLE
Bump Parthenon to latest develop (2024-02-15)

### PR DESCRIPTION
**Pull request by forrestglines**
Update Parthenon to PR930

### CODE REVIEW

**Comment by forrestglines** [comment]
@par-hermes format

**Comment by forrestglines** [comment]
Actually this causing issues for the cluster set up, do no merge

**Comment by pgrete** [comment]
FYI I just bumped to current `develop` in Parthenon. Let's see how this works.

**Comment by BenWibking** [comment]
I'm a bit confused as to why this is merging into the `donut` branch. The main branch doesn't have these changes either, does it?

Is this PR still relevant, or has it been superceded?

**Comment by BenWibking** [comment]
I think the core dump will point to that because that's where the error handler is, not necessarily because the underlying bug is there.

(We've been debugging an issue in Quokka that produces that error message. We thought it was a out-of-bounds memory access, which it usually is. However, It turned out to be a compiler bug that appears for specific ROCm versions 🙃)

Yes, that plan sounds good to me.

**Comment by BenWibking** [comment]
@pgrete just want to check if this is ready to go?

**Comment by pgrete** [comment]
> @pgrete just want to check if this is ready to go?

I double checked on Friday that the (other) crashes I observed are not reproducible any more. They were also not reproducible with the binary I triggered them in first place and given that those were all interconnect related issue, I assume that the machine was unstable by chance when I ran the original tests. (Note this was a second error mode apart from the `Memory access fault by GPU` above).

I now updated to latest Parthenon develop and fixed all interface changes *fingerscrossed*.
I also took this as opportunity to introduce a CHANGELOG (where I highlighted the changes in the Parthenon API).

Bottom line: this is ready for review and merge @BenWibking @forrestglines

**Comment by pgrete** [comment]
> There appears to be some unrelated cluster setup changes (although they are trivial). Everything else looks good to me.

Yes, I added those (to be used for the histograms).
In general, I'd like to separate the "derived fields" into a more central location as many fields are reusable across problem generators.

**Comment by BenWibking** [comment]
@pgrete do you want to merge this now?
